video/out/vo_dmabuf_wayland.c

```diff
@@ -692,6 +699,11 @@ static int reconfig(struct vo *vo, struct mp_image *img)
     if (!vo_wayland_reconfig(vo))
         return VO_ERROR;
 
+    mp_mutex_lock(&vo->params_mutex);
+    talloc_free(vo->target_params);
+    vo->target_params = talloc_dup(vo, &img->params);
```

Comment:
Do you think we need this to be dynamic allocation or we could keep it in priv?
``` c
p->target_params = img->params;
vo->target_params = &p->target_params;
```

Reply:
I think it has to be dynamic to be safer. Otherwise we are dependent on `mpctx->next_frames` to never ever free that particular img if we need the params. I'm not sure if that can happen but `run_reconfig` makes a copy for probably the same reason.

Reply:
Yes, but what I'm thinking is to put it into vo_dmabuf context, should be alive all the time.

``` diff
diff --git a/video/out/vo_dmabuf_wayland.c b/video/out/vo_dmabuf_wayland.c                                                                            index a0a8036848..ef123df071 100644                                                                                                                   --- a/video/out/vo_dmabuf_wayland.c                                                                                                                   +++ b/video/out/vo_dmabuf_wayland.c                                                                                                                   @@ -102,6 +102,8 @@ struct priv {                                                                                                                          enum hwdec_type hwdec_type;                                                                                                                           uint32_t drm_format;                                                                                                                                  uint64_t drm_modifier;                                                                                                                           +                                                                                                                                                     +    struct mp_image_params target_params;                                                                                                             };                                                                                                                                                                                                                                                                                                          static void buffer_handle_release(void *data, struct wl_buffer *wl_buffer)                                                                           @@ -699,8 +701,8 @@ done:                                                                                                                                      return VO_ERROR;                                                                                                                                                                                                                                                                                        mp_mutex_lock(&vo->params_mutex);                                                                                                                -    talloc_free(vo->target_params);                                                                                                                  -    vo->target_params = talloc_dup(vo, &img->params);                                                                                                +    p->target_params = img->params;                                                                                                                  +    vo->target_params = &p->target_params;                                                                                                                mp_mutex_unlock(&vo->params_mutex);                                                                                                                                                                                                                                                                         wl_surface_set_buffer_transform(vo->wl->video_surface, img->params.rotate / 90);
```

Reply:
Yeah I guess that works.